chaos_genius/app.py

```diff
@@ -114,6 +118,25 @@ def shell_context():
     app.shell_context_processor(shell_context)
 
 
+def register_hooks(app):
+    """Register hooks for Flask requests."""
+    @app.before_request
+    def demo_endpoints_blocker():
+        """Hook to allow access to only whitelisted APIs in demo."""
+        if REACT_APP_IS_DEMO:
+            @app.before_request
+            def before_request():
+                if request.endpoint not in DEMO_ENDPOINT_WHITELIST:
+                    return jsonify({"status": "failure",
+                                    "message": "Endpoint not accessible in demo version"})
+                else:
+                    if request.method not in DEMO_ENDPOINT_WHITELIST[request.endpoint]:
+                        return jsonify({"status": "failure",
+                                        "message": f"{request.method} method is not allowed for this endpoint in demo version"})
```

Comment:
This needs formatting:
```suggestion
                    return jsonify(
                        {
                            "status": "failure",
                            "message": "Endpoint not accessible in demo version"
                        }
                    )
                else:
                    if request.method not in DEMO_ENDPOINT_WHITELIST[request.endpoint]:
                        return jsonify(
                            {
                                "status": "failure",
                                "message": f"{request.method} method is not allowed "
                                "for this endpoint in demo version"
                            }
                        )

```